test/integration/test-submit-url-parsing.js

```diff
@@ -0,0 +1,27 @@
+var http = require('http');
+var https = require('https');
+var common = require('../common');
+var assert = common.assert;
+var FormData = require(common.dir.lib + '/form_data');
+
+/**
+ * Test url parsing during submission
+ */
+
+var req;
+var form = new FormData();
+form.append('field', 'value');
+
+// Basic parsing
+req = form.submit('http://localhost:' + common.port + '/path', function() {});
+assert.strictEqual(req.path, '/path');
+assert.ok(req.agent instanceof http.Agent, 'req.agent instanceof http.Agent');
+assert.strictEqual(req.getHeader('Host'), 'localhost:' + common.port);
```

Comment:
Can you add similar check to the https counter part?
And maybe try with default ports, since you're going through with the request anyway.

Sounds good?